### ISSUE TITLE
way to use any other gateway without using default gateway in the computer

### ISSUE THREAD

**Issue by asinthe1** [justcoding121/Titanium-Web-Proxy#779]
Is there any option to use another gateway for the proxy server without using the default gateway in windows

thanks

**Comment by honfika** [comment]
Yes, you can set any upstream proxy... you don't have to use the windows proxy settings. Moreover you can use different upsteream proxy for each request (for example by url or for load balancing reason)

**Comment by asinthe1** [comment]
thanks for the quick reply
can you please direct me to sample code

thanks and best regards

**Comment by honfika** [comment]
You can find the sample code here:
https://github.com/justcoding121/Titanium-Web-Proxy/blob/master/examples/Titanium.Web.Proxy.Examples.Basic/ProxyTestController.cs

**Comment by asinthe1** [comment]
Thanks for your reply
i try like this 
var proxyServer = new ProxyServer();
proxyServer.UpStreamHttpProxy = new ExternalProxy("192.168.1.2", 808);
proxyServer.UpStreamHttpsProxy = new ExternalProxy("192.168.1.2", 808);

its works fine for all traffic but if i want external proxy for specific sites for eg: facebook.com
how to implement that 
i try OnRequest method like this
if (e.HttpClient.Request.RequestUri.AbsoluteUri.Contains("facebook.com"))
{
                e.CustomUpStreamProxy = new ExternalProxy("192.168.1.12", 808);
}

then its not working is there any option to do that

thank you very much
 and best regards

**Comment by honfika** [comment]
Is your code at least called?

**Comment by asinthe1** [comment]
yes its called i try in debug mode and object e.CustomUpStreamProxy has new value as well

**Comment by justcoding121** [comment]
Closing this as a networking/configuration question. Titanium can select an upstream HTTP/SOCKS proxy, including per request through GetCustomUpStreamProxyFunc. It cannot directly select an IP-layer default gateway; that is controlled by the operating system routing table or by binding/routing a network interface. For host-based proxy selection, return the desired ExternalProxy from GetCustomUpStreamProxyFunc and return null for direct connections.
